lib/game.rb

```diff
@@ -0,0 +1,90 @@
+require_relative 'display'
+require_relative 'player'
+require_relative 'grid'
+require_relative 'converter'
+require_relative 'move_validator'
+
+class Game
+
+  def initialize(display, player, grid, move_validator)
+    @display = display
+    @player = player
+    @grid = grid
+    @move_validator = move_validator
+  end
+
+  def only_get_valid_letters
+    letter = @player.get_letter_coordinate
+    until @move_validator.is_letter_valid?(letter)
+      @display.ask_for_letter
+      letter = @player.get_letter_coordinate
+    end
+    letter
+  end
+
+  def only_get_valid_numbers
+    number = @player.get_number_coordinate
+    until @move_validator.is_number_valid?(number)
+      @display.ask_for_number
+      number = @player.get_number_coordinate
+    end
+    number
+  end
+
+  def move_played_before?(move, past_moves)
+    past_moves.include?(move)
+  end
+
+  def ship_coordinates(move, ship_coordinates)
+    ship_coordinates.include?(move)
+  end
+
+  def new_move
+    @display.ask_for_number
+    x_coordinate = only_get_valid_numbers
+    @display.ask_for_letter
+    y_coordinate = only_get_valid_letters
+    [x_coordinate, y_coordinate]
+  end
+
+  def convert_move(x_coordinate, y_coordinate)
+    converter = Converter.new
+    array_position_1 = converter.number_to_array_position(x_coordinate)
+    array_position_2 = converter.letter_to_array_position(y_coordinate)
+    [array_position_1, array_position_2]
+  end
+
+  def game_flow(grid_size, ship_coordinates)
+    grid = @grid.draw_grid(grid_size)
+    mark = "X"
+    past_moves = []
+    coordinates_list = ship_coordinates.length
+    @display.display_grid(grid_size)
```

Comment:
See comment above in Display class on line 10, where you define the message `display_grid`. In this Game class you already have access to `@grid`, you could pass it into the `display_grid` method here.